daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -443,7 +443,13 @@ private[lf] object SBuiltin {
       val func = args.get(0)
       val init = args.get(1)
       val list = args.get(2).asInstanceOf[SList].list
-      machine.pushKont(KFoldl(func, list, machine.frame, machine.actuals, machine.env.size))
+      machine.pushKont(
```

Comment:
Don't we need to pop from the environment here as well to enable tail call optimization for `foldl`? Either way, let's make sure we add a test to ensure that `foldl` does indeed not consume stack in the follow-up PR containing the tests for TCO.

Reply:
I dont think so.
- The pop will occur when we process the return value `machine.returnValue = init` in the top level `run` loop, on the next step.
- (And we didn't call `markBase` when we pushed this continuation.)

Reply:
Either way. Let's make sure we have tests showing this runs in constant stack space.

Reply:
:+1: